elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,247 @@
+         break;
+       case KEYCODE.ESCAPE:
+         // If escape, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       case KEYCODE.TAB:
+         // If tab, close the menu.
+         event.preventDefault();
+         this.opened = false;
+         break;
+       default:
+         break;
+      }
+      // If letter key, move to an item which starts with that letter.
+      if (event.keyCode >= KEYCODE.A && event.keyCode <= KEYCODE.Z) {
+        for (let child of this.children) {
+          if (child.innerText.trim()[0] === event.key) {
+            event.preventDefault();
+            child.focus();
+            break;
+          }
+        }
+      }
+    }
+
+    open() {
+      this._addTabindex();
+      let first = this._firstMenuItem();
+      first && first.focus();
+    }
+
+    set opened(isOpened) {
+      if (!!isOpened) {
+        this.setAttribute('opened', 'true');
+        this.open();
+      } else {
+        this.removeAttribute('opened');
+        this._removeTabindex();
+      }
+    }
+
+    get opened() {
+      return this.hasAttribute('opened');
+    }
+
+    _upgradeProperty(prop) {
+      if (this.hasOwnProperty(prop) || this.hasAttribute(prop)) {
+        let value = this[prop];
+        delete this[prop];
+        this[prop] = value;
+      }
+    }
+
+    _onFocusOut(event) {
+      if (event.relatedTarget === null || !this.contains(event.relatedTarget)) {
+        this.opened = false;
+      }
+    }
+
+    /**
+     * `connectedCallback` fires when the element is inserted into the DOM.
+     * It's a good place to set the initial `role`, `tabindex`, internal state,
+     * and install event listeners.
+     */
+    connectedCallback() {
+      if (!this.hasAttribute('role')) {
+        this.setAttribute('role', 'menu');
+      }
+
+      // A user may set a property on an _instance_ of an element,
+      // before its prototype has been connected to this class.
+      // The `_upgradeProperty` method will check for any instance properties
+      // and run them through the proper class setters.
+      // See the [lazy properites](#lazy-properties) section for more details.
+      this._upgradeProperty('opened');
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('focusout', this._onFocusOut);
+    }
+
+    /**
+     * Unregisters the event listeners that were set up in `connectedCallback`.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('focusout', this._onKeyDown);
+    }
+
+    /**
+     * `attributeChangedCallback` is called when any of the attributes in the
+     * `observedAttributes` array are changed. It's a good place to handle
+     * side effects, like changing the state of the widget.
+     */
+    attributeChangedCallback(name, oldValue, newValue) {
+      if (name === 'opened') {
+        if (!!newValue) {
```

Comment:
Do you mean a switch on "name" variable? It seems kinda weird to switch in case of only 1 value ... Or switch on "newValue"? But then switch on a true/false also seems weird.